integration_tests/dataplane/mymac/mymac_test.go

```diff
@@ -189,65 +186,147 @@ func clearMyMac(t *testing.T, dut *ondatra.DUTDevice) error {
-	conn := dataplaneConn(t, dut)
-	mmc := saipb.NewMyMacClient(conn)
-	_, err := mmc.CreateMyMac(context.Background(), &saipb.CreateMyMacRequest{
+	resp, err := mmc.CreateMyMac(context.Background(), &saipb.CreateMyMacRequest{
 		Switch:         1,
 		Priority:       proto.Uint32(1),
 		MacAddress:     []byte{0, 0, 0, 0, 0, 0},
 		MacAddressMask: []byte{0, 0, 0, 0, 0, 0},
 	})
 	if err != nil {
-		return err
+		return 0, err
 	}
 	t.Logf("MyMac table restored.")
+	return resp.GetOid(), nil
+}
+
+// printMyMacEntries prints the entries in the MyMac table.
+func printMyMacEntries(t *testing.T, sc saipb.SwitchClient, mmc saipb.MyMacClient) error {
+	var err error
+	req := &saipb.GetSwitchAttributeRequest{
+		Oid:      1,
+		AttrType: []saipb.SwitchAttr{saipb.SwitchAttr_SWITCH_ATTR_MY_MAC_LIST},
+	}
+	resp, err := sc.GetSwitchAttribute(context.Background(), req)
+	if err != nil {
+		return err
+	}
+	oids := resp.GetAttr().MyMacList
+	if oids == nil {
+		return nil
+	}
+	for _, oid := range oids {
+		resp2, err := mmc.GetMyMacAttribute(context.Background(), &saipb.GetMyMacAttributeRequest{
+			Oid: oid,
+		})
+		if err != nil {
+			return err
+		}
+		t.Logf(">>>>>> MyMac[%d]: %+v", oid, resp2.GetAttr())
+	}
 	return nil
 }
 
+// entry contains the MyMac request information.
+type entry struct {
+	priority    uint32
+	macAddr     string
+	macAddrMask []byte
+}
+
+type myMacTest struct {
+	desc       string
+	clearMyMac bool
+	dstMAC     string
+	entries    []entry
+	passed     bool
+}
+
 func TestMyMac(t *testing.T) {
-	// TODO: add more sub-tests when the Magna issue is resolved.
-	tests := []struct {
-		desc        string
-		clearMyMac  bool
-		macAddr     []byte
-		macAddrMask []byte
-		passed      bool
-	}{{
-		desc:       "Traffic dropped", // Remove the default entry that allows all traffic to L3.
+	specialMAC := "00:1A:11:17:5F:80"
+	tests := []myMacTest{{
+		desc:       "Traffic passed by default",
+		clearMyMac: false,
+		passed:     true,
+	}, {
+		desc:       "Traffic dropped with clearing MyMac table", // Remove the default entry that allows all traffic to L3.
 		clearMyMac: true,
 		passed:     false,
+	}, {
+		desc:       "Traffic passed with specific allowed MAC address",
+		clearMyMac: true,
+		entries: []entry{{
+			priority:    2010,
+			macAddr:     dutPort1.MAC,
+			macAddrMask: []byte{0xff, 0xff, 0xff, 0xff, 0xff, 0xff},
+		}, {
+			priority:    2000,
+			macAddr:     specialMAC,
+			macAddrMask: []byte{0xff, 0xff, 0xff, 0xff, 0xff, 0xff},
+		}},
+		passed: true,
+	}, {
+		desc:       "Traffic dropped with specific allowed MAC address",
+		clearMyMac: true,
+		entries: []entry{{
+			priority:    2010,
+			macAddr:     dutPort2.MAC,
+			macAddrMask: []byte{0xff, 0xff, 0xff, 0xff, 0xff, 0xff},
+		}, {
+			priority:    2000,
+			macAddr:     specialMAC,
+			macAddrMask: []byte{0xff, 0xff, 0xff, 0xff, 0xff, 0xff},
+		}},
+		passed: false,
+	}, {
+		desc:       "Traffic passed with special MAC address",
```

Comment:
nit: special is not descriptive, something like  "passed where dst mac not interface mac"

Reply:
Done